src/player.cpp

```diff
@@ -2097,14 +2099,28 @@ Item* Player::getCorpse(Creature* lastHitCreature, Creature* mostDamageCreature)
 {
 	Item* corpse = Creature::getCorpse(lastHitCreature, mostDamageCreature);
 	if (corpse && corpse->getContainer()) {
-		std::ostringstream ss;
+		std::unordered_map<std::string, uint16_t> names;
+		for (const auto& killer : getKillers()) {
+			++names[killer->getName()];
+		}
+
 		if (lastHitCreature) {
-			ss << "You recognize " << getNameDescription() << ". " << (getSex() == PLAYERSEX_FEMALE ? "She" : "He") << " was killed by " << lastHitCreature->getNameDescription() << '.';
+			if (!mostDamageCreature) {
+				corpse->setSpecialDescription(fmt::format("You recognize {:s}. {:s} was killed by {:s}{:s}", getNameDescription(), getSex() == PLAYERSEX_FEMALE ? "She" : "He", lastHitCreature->getNameDescription(), names.size() > 1 ? " and others." : "."));
+			} else {
+				if (lastHitCreature != mostDamageCreature && names[lastHitCreature->getName()] == 1) {
+					corpse->setSpecialDescription(fmt::format("You recognize {:s}. {:s} was killed by {:s} and {:s}{:s}", getNameDescription(), getSex() == PLAYERSEX_FEMALE ? "She" : "He", mostDamageCreature->getNameDescription(), lastHitCreature->getNameDescription(), names.size() > 2 ? " and others." : "."));
+				} else {
+					corpse->setSpecialDescription(fmt::format("You recognize {:s}. {:s} was killed by {:s} and others.", getNameDescription(), getSex() == PLAYERSEX_FEMALE ? "She" : "He", mostDamageCreature->getNameDescription()));
+				}
+			}
 		} else {
-			ss << "You recognize " << getNameDescription() << '.';
+			if (mostDamageCreature) {
+				corpse->setSpecialDescription(fmt::format("You recognize {:s}. {:s} was killed by something evil and {:s}{:s}", getNameDescription(), getSex() == PLAYERSEX_FEMALE ? "She" : "He", mostDamageCreature->getNameDescription(), names.size() > 1 ? " and others." : "."));
```

Comment:
This phrase is a bit weird. When names.size() > 1, it would write:

> You recognize Player Name. He was killed by something evil and Killer Name and others.

While I believe it would be more correct to write

> You recognize Player Name. He was killed by something evil, Killer Name and others.

```suggestion
				if (names.size() > 1) {
					corpse->setSpecialDescription(fmt::format("You recognize {:s}. {:s} was killed by something evil, {:s} and others;", getNameDescription(), getSex() == PLAYERSEX_FEMALE ? "She" : "He", mostDamageCreature->getNameDescription()));
				} else {
					corpse->setSpecialDescription(fmt::format("You recognize {:s}. {:s} was killed by something evil and {:s}.", getNameDescription(), getSex() == PLAYERSEX_FEMALE ? "She" : "He", mostDamageCreature->getNameDescription()));
				}
```

Reply:
Great diffing skills, Github :+1: